hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c

```diff
@@ -791,7 +791,7 @@ static int create_container_directories(const char* user, const char *app_id,
     if (container_dir == NULL) {
       return OUT_OF_MEMORY;
     }
-    if (strstr(container_dir, "..") != 0) {
```

Comment:
could they also be '/..' and not end with '/'

Reply:
Theoretically, yes. Practically speaking, when I looked at the components of the strings, the ends weren't user controllable. but I can implement a `strtok_r` variant that would handle that case.

Reply:
Ok, that wasn't so bad: https://gist.github.com/ddelong/c5485610a46b4257c5f5f4df2d5fcc8f

I'll update this.